pkg/provider/handlers/secret_test.go

```diff
@@ -161,3 +163,69 @@ func TestSecretCreation(t *testing.T) {
 		})
 	}
 }
+
+func TestListSecrets(t *testing.T) {
+	mountPath, err := os.MkdirTemp("", "test_secret_creation")
+	if err != nil {
+		t.Fatalf("unexpected error while creating temp directory: %s", err)
+	}
+
+	defer os.RemoveAll(mountPath)
+
+	handler := MakeSecretHandler(nil, mountPath)
+
+	cases := []struct {
+		name       string
+		verb       string
+		status     int
+		secretPath string
+		secret     string
+		err        string
+		expected   []types.Secret
+	}{
+		{
+			name:       "Get empty secret list for namespace having no secret",
+			verb:       http.MethodGet,
+			status:     http.StatusOK,
```

Comment:
The test case doesn't account for the non-default namespace, i think it needs another field `namespace` to accurately test the bug I saw. This first case is fine, it represents the default namespace, but it seems like we need a second case for a non-default namespace.  If i recall correctly, it should be as simple as adding `"?namespace="+tc.namespace` to the url _if_ `tc.namespace` is not empty.

Reply:
I could not find any example to mock containerd client. `listSecrets` method calls `validNamespace` method, which need uses containerd client.

https://github.com/openfaas/faasd/blob/acb5d0bd1cc5876d05261857b0717b779b95ec1b/pkg/provider/handlers/secret.go#L53

Reply:
@LucasRoesler Have made the changes as per comment in test.